core/src/services/gdrive/builder.rs

```diff
@@ -23,41 +23,63 @@ use std::sync::Arc;
 use chrono::DateTime;
 use chrono::Utc;
 use log::debug;
+use serde::Deserialize;
 use tokio::sync::Mutex;
 
 use super::backend::GdriveBackend;
-use crate::raw::HttpClient;
 use crate::raw::{normalize_root, PathCacher};
+use crate::raw::{ConfigDeserializer, HttpClient};
 use crate::services::gdrive::core::GdriveSigner;
 use crate::services::gdrive::core::{GdriveCore, GdrivePathQuery};
 use crate::Scheme;
 use crate::*;
 
+/// [GoogleDrive](https://drive.google.com/) configuration.
+#[derive(Default, Deserialize)]
+#[serde(default)]
+#[non_exhaustive]
+pub struct GdriveConfig {
+    /// The root for gdrive
+    pub root: Option<String>,
+    /// Access token for gdrive.
+    pub access_token: Option<String>,
+    /// Refresh token for gdrive.
+    pub refresh_token: Option<String>,
+    /// Client id for gdrive.
+    pub client_id: Option<String>,
+    /// Client secret for gdrive.
+    pub client_secret: Option<String>,
```

Comment:
Maybe add an option for permanent remove or temporary trash by default?

And only judging by config is kind of too dull, we may also extend OpDelete with an optional field, as an override to the config.

Reply:
> Maybe add an option for permanent remove or temporary trash by default?

We can add an option for permanent deletion if users request it in the future.

> And only judging by config is kind of too dull, we may also extend OpDelete with an optional field, as an override to the config.

I feel like this idea is not cool since not all services have such concept.